packages/react-components/src/components/chart/legend/tanstackTable.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+
+import {
+  spaceStaticXxs,
+  colorBorderButtonPrimaryDisabled,
+  colorBackgroundButtonNormalDefault,
+  spaceStaticM,
+  spaceStaticXxxs,
+  colorBackgroundLayoutToggleActive,
+  spaceStaticXs,
+} from '@cloudscape-design/design-tokens';
+
+import {
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+  getSortedRowModel,
+  SortingState,
+  ColumnDef,
+} from '@tanstack/react-table';
+
+import './tanstackTable.css';
+import { sortingIconAsc, sortingIconDesc } from './sotingIcons';
+
+type tanstackTableOptions = {
+  data: object[];
+  columnDefinitions: ColumnDef<object, string>[];
+  stickyColumns?: { first: number };
+  stickyHeader?: boolean;
+};
+
+export const TanstackTable = ({ data, columnDefinitions, stickyColumns, stickyHeader }: tanstackTableOptions) => {
```

Comment:
Overall implementation of <TanstackTable /> component looks good. @Chandru-HM how was the implementation for you? Straight-forward or did you run into any issues?

Reply:
Thank you.
It's good, more customizable and Straight-forward. but for few I made custom, due to no default property is available (like sticky header, sticky column and sorting)